lib/Sema/TypeCheckPattern.cpp

```diff
@@ -768,13 +768,27 @@ ExprPatternMatchRequest::evaluate(Evaluator &evaluator,
       DeclNameLoc(EP->getLoc()));
   matchOp->setImplicit();
 
+  auto subExpr = EP->getSubExpr();
+
+  // Pull off the outer "unsafe" expression.
+  UnsafeExpr *unsafeExpr = dyn_cast<UnsafeExpr>(subExpr);
+  if (unsafeExpr) {
+    subExpr = unsafeExpr->getSubExpr();
+  }
+
```

Comment:
Can `UnsafeExpr` be wrapped in an `IdentityExpr`?

Reply:
So, that is broken, but I'm going to file a follow-up bug for myself because the `try unsafe` case is annoyingly hard to fix without breaking internal invariants.